post-processor/vagrant/post-processor.go

```diff
@@ -292,6 +301,10 @@ func (p *PostProcessor) configureSingle(c *Config, raws ...interface{}) error {
 	}
 
 	var errs *packersdk.MultiError
+	if c.VagrantfileTemplateContent != "" && c.VagrantfileTemplate == "" {
+		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf(
+			"'vagrantfile_template' should be provided to use 'vagrantfile_template_content'"))
+	}
```

Comment:
I of the understanding that you would want to specify just the content with no file path needed. Is that not the case? 

If they are mutually exclusive I would the logic to be as follows: 
```suggestion
	if c.VagrantfileTemplateContent != "" && c.VagrantfileTemplate != "" {
		errs = packersdk.MultiErrorAppend(errs, fmt.Errorf(
			"'vagrantfile_template' can not be used with 'vagrantfile_template_content'"))
	}
```

Reply:
@nywilken if you see the example [here](https://github.com/hashicorp/packer-plugin-vagrant/pull/112#issue-2116501876) these are Not Exclusive we are trying to load  the value for the variable `my_ssh_key` in the `vagrant_template`